### PR TITLE
Clarify local development setup

### LINKED ISSUE #1874: Seeking faster ways to open the frontend
Hi,
Every time I want to add a new component in the frontend and see the updates in the website, the only method I know now is:
1. Add new components
2. Manually run `npm run build`
3. Manually run `cargo build --bin site --release`
4. and Run the service
which is lengthy and crazy.

I wonder if there's a way to quickly open just the frontend. Typically, we'd have something like `npm run dev` for this. But I did not see any here 🫠

### PR DESCRIPTION
**Pull request by chengr4**
Close #1874 

Please confirm whether the update is appropriate.

### CODE REVIEW

**Review by Kobzol — COMMENTED** [review]
It doesn't incorporate it, you still need to execute `cargo build` or `cargo run` manually. The important info is that if you want to develop locally, you should use watch + cargo run, without `--release`.

**Comment by chengr4** [comment]
How about this?

File: site/frontend/README.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ This directory contains code for the `rustc-perf` website frontend.
     ```console
     $ npm run watch
     ```
+    > If you want to develop locally, it's recommended to use the `npm run watch` + `cargo run --bin site` (without the `--release` flag). This setup enables hot reloading of your code on the website.
```
Comment (Kobzol) [inline site/frontend/README.md:13]:
```suggestion
    > If you want to develop locally, it's recommended to use `npm run watch` + `cargo run --bin site` (without the `--release` flag). This setup enables hot reloading of your code on the website.
```

**Review by Kobzol — APPROVED** [review]
There was one extra word, otherwise LGTM.

File: site/frontend/README.md (hunk 1 above)
Comment (chengr4) [reply site/frontend/README.md]: fixed it!

**Review by Kobzol — APPROVED** [review]
Thanks :)

**Pull request merged**